grammar.js

```diff
@@ -608,6 +609,36 @@ module.exports = grammar({
 
     pointer_type: $ => prec(PREC.POSTFIX, seq($._type, '*')),
 
+    function_pointer_type: $ => seq(
+      'delegate',
+      '*',
+      optional($.calling_convention_specifier),
+      '<',
+      commaSep1($.funcptr_parameter),
+      '>'
+    ),
+
+    calling_convention_specifier: $ => choice(
+      'managed',
+      seq(
+        'unmanaged',
+        optional(seq('[', $.unmanaged_calling_convention, ']'))
+      )
+    ),
+
+    unmanaged_calling_convention: $ => choice(
+      'Cdecl',
+      'Stdcall',
+      'Thiscall',
+      'Fastcall',
+      commaSep1($.identifier)
+    ),
+
+    funcptr_parameter: $ => seq(
```

Comment:
Roslyn calls this `FunctionPointerParameter` so perhaps `function_pointer_parameter` ?

Reply:
Also think we're missing corpus coverage for this.